### PR TITLE
Add new MCA parameter to support edge case with debugger at LLNL

### PR DESCRIPTION
**Pull request by rhc54**
@jsquyres please review

### CODE REVIEW

**Comment by mellanox-github** [comment]
- Coverity found 1488 errors for all: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/all_168/output/errors/index.html
- Coverity found 29 errors for oshmem: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/oshmem_168/output/errors/index.html
- Coverity found 7 errors for fca: http://bgate.mellanox.com:8888/jenkins/job/gh-ompi-release-v1.8-pr//ws/cov_build/fca_168/output/errors/index.html <br><br>:octocat: _Sent from [GH](http://nodegh.io)._

**Comment by mellanox-github** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/168/
Test PASSed.

File: orte/mca/ras/slurm/ras_slurm_component.c [hunk 1]
```diff
@@ -106,6 +106,14 @@ static int ras_slurm_register(void)
                                             MCA_BASE_VAR_SCOPE_READONLY,
                                             &mca_ras_slurm_component.rolling_alloc);
 
+    mca_ras_slurm_component.use_all = false;
+    (void) mca_base_component_var_register (component, "use_entire_allocation",
+                                            "Use entire allocation (not just job step nodes) for this application",
+                                            MCA_BASE_VAR_TYPE_BOOL, NULL, 0, 0,
+                                            OPAL_INFO_LVL_9,
```
Comment (jsquyres) [inline orte/mca/ras/slurm/ras_slurm_component.c]: My only nit is that I wouldn't make this level 9 -- perhaps 4 or 5?  (I don't fully grok the use case, so I don't know exactly who will be using this variable -- see https://github.com/open-mpi/ompi/wiki/MCAParamLevels)

Comment (rhc54) [inline orte/mca/ras/slurm/ras_slurm_component.c]: I confess that I've given up on the levels, Jeff. There are just too many of them, and nobody really understands them in the user community. I'll put it to a 5 just to set it to something, but we really should address this someday.

**Pull request merged**